.github/workflows/pre-merge-rust.yml

```diff
@@ -90,3 +90,28 @@ jobs:
       working-directory: ${{ matrix.dir }}
       # NOTE: --all-targets doesn't run doc tests
       run: cargo test --locked --all-targets
+
+  block-manager-rust:
+    runs-on:
+      group: Fastchecker
+    name: Docker Rust Checks (llm/block-manager)
```

Comment:
Thanks for moving this!

Please add an inline comment motivating why this step needs to be run in docker for now (ex: requires specific NIXL libs/deps). 

We should probably try to work with nixl team to fix/validate the non-docker path with a "pip install nixl" or something since we advertise support for a pip install only path (no container).